index.js

```diff
@@ -16,11 +16,28 @@ const BAD_YAML_MESSAGE =
  * @example test -> Test
  * @example test text -> TestText
  * @example test-text -> TestText
+ * @example TESTTest -> TESTTest
  * @param {string} string - input string
  * @returns a string in PascalCase
  */
 const pascalCase = (string) => {
-	return _.upperFirst(_.camelCase(string));
+	// First, split the string into words using "look-ahead" & remove whitespace
+	const words = string.split(/(?=[A-Z])|[\s-_]+/).filter((word) => word.length > 0);
+
+	return words
```

Comment:
I sort of question the value of over complicating this function for what may amount to a conventional personal preference

Reply:
what situation *requires* that a function name be CAPITALthenlower?

Reply:
We're accounting for a feature that gives us names with acronyms in uppercase, like `USAServiceStuff` (from server team) and the current function logic changes it to `UsaServiceStuff`. Our code base currently is unable to find the definition for `VcrServiceStuff` because we're expecting `USAServiceStuff`.

Reply:
After looking, I guess acronyms stay all caps in pascal case, though it also seems debated.